data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java

```diff
@@ -122,34 +125,36 @@ public double getMaxValue(Network network) {
     @Override
     protected double getMinValueWithRange(Network network, Range range) {
         double minValue = range.getMin();
-        return convertTapToAngle(network, Math.max(lowTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), minValue)));
+        return convertTapToAngle(Math.max(lowTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), minValue)));
     }
 
     @Override
     protected double getMaxValueWithRange(Network network, Range range) {
         double maxValue = range.getMax();
-        return convertTapToAngle(network, Math.min(highTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), maxValue)));
+        return convertTapToAngle(Math.min(highTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), maxValue)));
     }
 
     @Override
     public double getMaxNegativeVariation(Network network) {
         // This method calls getMinValue so it will throw a NotSynchronizedException if required
-        return Math.max(convertTapToAngle(network, getCurrentTapPosition(network)) - getMinValue(network), 0);
+        return Math.max(convertTapToAngle(getCurrentTapPosition(network)) - getMinValue(network), 0);
     }
 
     @Override
     public double getMaxPositiveVariation(Network network) {
         // This method calls getMaxValue so it will throw a NotSynchronizedException if required
-        return Math.max(getMaxValue(network) - convertTapToAngle(network, getCurrentTapPosition(network)), 0);
+        return Math.max(getMaxValue(network) - convertTapToAngle(getCurrentTapPosition(network)), 0);
     }
 
     private int getCurrentTapPosition(Network network) {
-        return network.getTwoWindingsTransformer(networkElement.getId()).getPhaseTapChanger().getTapPosition();
+        return checkValidPstAndGetPhaseTapChanger(network).getTapPosition();
```

Comment:
you could do like you did with convertTapToAngle and remove the network argument from this method

Reply:
As it is the current position that we want in a specified network I think it is better to keep the network as argument. Otherwise it will return the position of the synchronized network.

Reply:
I believe the phase tap changer's getTapPosition gets the tap position for the "current" variant of the network. But this method is being changed in another PR so np